src/quick-lint-js/parse.h

```diff
@@ -2516,7 +2516,18 @@ class parser {
 
     if (this->peek().type == token_type::kw_else) {
       this->skip();
+
+      bool semicolon_after_else = this->peek().type == token_type::semicolon;
+      const char8 *token_after_else_begin = this->peek().begin;
```

Comment:
Wanted to avoid copying the token. Maybe its premature optimization on my part.

Reply:
Copying a `token` should be cheap thanks for Scalar Replacement of Aggregates optimizations. (After optimization, only fields which are used are copied, which basically has the same effect as your manually-optimized code.)